lib/src/license_detection/new_tokenizer.dart

```diff
@@ -0,0 +1,256 @@
+import 'dart:collection';
+
+import 'package:meta/meta.dart';
+import 'package:source_span/source_span.dart';
+import 'package:string_scanner/string_scanner.dart';
+
+@sealed
+class Token {
+  /// Normalized form of the text in [fileSpan].
+  final String value;
+
+  /// Denotes the token position.
+  final int tokenID;
+
+  /// Zero based line number of token.
+  final int line;
+
+  /// SourceSpan of the token.
+  final SourceSpan fileSpan;
+
+  Token(this.value, this.tokenID, this.line, this.fileSpan);
+
+  Token.fromSpan(this.fileSpan, this.tokenID)
+      : value = _normalizeWord(fileSpan.text.toLowerCase()),
+        line = fileSpan.start.line;
+}
+
+List<Token> tokenizer(String text) {
+  final _scanner = SpanScanner(text);
+
+  var tokens = <Token>[];
+  var tokenID = 0;
+
+  Token? nextToken() {
+    if (_scanner.scan(_wordRegex)) {
```

Comment:
So `hello world!` becomes `hello` and `world!` are you sure we want `world!` and not `world`? I don't know I'm just asking :D